x/mint/keeper/integration_test.go

```diff
@@ -64,8 +74,86 @@ func genApp(withGenesis bool, invCheckPeriod uint) (*spnapp.App, spnapp.GenesisS
 
 	originalApp := app.(*spnapp.App)
 	if withGenesis {
-		return originalApp, spnapp.NewDefaultGenesisState(encCdc.Marshaler)
+		genesisState := spnapp.NewDefaultGenesisState(encCdc.Marshaler)
+		privVal := mock.NewPV()
+		pubKey, _ := privVal.GetPubKey()
+		// create validator set with single validator
+		validator := tmtypes.NewValidator(pubKey, 1)
+		valSet := tmtypes.NewValidatorSet([]*tmtypes.Validator{validator})
+
+		// generate genesis account
+		senderPrivKey := secp256k1.GenPrivKey()
+		acc := authtypes.NewBaseAccount(senderPrivKey.PubKey().Address().Bytes(), senderPrivKey.PubKey(), 0, 0)
+		balances := banktypes.Balance{
+			Address: acc.GetAddress().String(),
+			Coins:   sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100000000000000))),
+		}
+		genesisState = genesisStateWithValSet(encCdc.Marshaler, genesisState, valSet, []authtypes.GenesisAccount{acc}, balances)
+		return originalApp, genesisState
 	}
 
 	return originalApp, spnapp.GenesisState{}
 }
+
+func genesisStateWithValSet(
```

Comment:
Is it a requirement for this upgrade or part of https://github.com/tendermint/spn/issues/852?

Looks like this utility could be moved to `testutil`

Reply:
This is required for the upgrade.  The test will now fail without a validator in the state.  I agree though - moving to `testutil`.